protocol/contracts/tokens/Fertilizer/Internalizer.sol

```diff
@@ -38,12 +40,69 @@ contract Internalizer is OwnableUpgradeable, ReentrancyGuardUpgradeable, Fertili
 
     string private _uri;
 
-    function uri(uint256 _id) external view virtual override returns (string memory) {
-        return string(abi.encodePacked(_uri, StringsUpgradeable.toString(_id)));
+    // ----------------------------- NEW URI FUNCTION ----------------------------
+
+    /**
+        * @notice Assembles and returns a base64 encoded json metadata
+        * URI for a given fertilizer ID.
+        * need to ovveride because the contract indirectly
+        * inherits from ERC1155
+        * @param _id - the id of the fertilizer
+        * @return - the json metadata URI
+     */
+    function uri(uint256 _id)
+        external
+        view
+        virtual
+        override 
+        returns (string memory)
+    {
+
+        uint128 bpfRemaining = calculateBpfRemaining(_id);
+
+        // generate the image URI
+        string memory imageUri = imageURI(_id , bpfRemaining);
+
+        // assemble and return the json URI
+        return (
+            string(
+                abi.encodePacked(
+                    BASE_JSON_URI,
+                    Base64.encode(
```

Comment:
any reason that we're using Base64.sol here instead of LibBytes64.sol used in MetadataImage.sol? Wondering if there is a benefit.

Reply:
I didn't realize we had an in house implementation of that. I guess there is a tool for everything in Beanstalk! Replaced all references [here](https://github.com/BeanstalkFarms/Beanstalk/pull/751/commits/c449f573a573dbf1b5677ccdfea3bbde3dee4381)